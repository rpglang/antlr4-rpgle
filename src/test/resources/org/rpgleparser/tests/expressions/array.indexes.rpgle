        arr(*) = field2 * filed3;
        cust(custom).account(accum).balance -= 24; 