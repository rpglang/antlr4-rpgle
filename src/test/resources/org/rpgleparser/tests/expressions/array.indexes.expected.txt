/*===TOKENS===*/
ID                       <arr>
'('                      <(>
ARRAY_REPEAT             <*>
')'                      <)>
'='                      <=>
ID                       <field2>
MULT                     <*>
ID                       <filed3>
FREE_SEMI                <;>
ID                       <cust>
'('                      <(>
ID                       <custom>
')'                      <)>
'.'                      <.>
ID                       <account>
'('                      <(>
ID                       <accum>
')'                      <)>
'.'                      <.>
ID                       <balance>
'-='                     <-=>
NUMBER                   <24>
FREE_SEMI                <;>
/*===TREE===*/
(r
  (statement
    (free
      (baseExpression
        (op
          (op_eval
            (evalExpression
              (assignmentExpression
                (simpleExpression
                  (identifier
                    (multipart_identifier
                      (indexed_identifier (free_identifier (idOrKeyword arr)) ( * ))
                    )
                  )
                )
                =
                (expression
                  (expression (identifier (free_identifier (idOrKeyword field2))))
                  *
                  (expression (identifier (free_identifier (idOrKeyword filed3))))
                )
              )
            )
          )
        )
      )
      ;
    )
  )
  (statement
    (free
      (baseExpression
        (op
          (op_eval
            (evalExpression
              (assignOperatorExpression
                (simpleExpression
                  (identifier
                    (multipart_identifier
                      (indexed_identifier
                        (free_identifier (idOrKeyword cust))
                        (
                        (expression (identifier (free_identifier (idOrKeyword custom))))
                        )
                      )
                      .
                      (indexed_identifier
                        (free_identifier (idOrKeyword account))
                        (
                        (expression (identifier (free_identifier (idOrKeyword accum))))
                        )
                      )
                      .
                      (free_identifier (idOrKeyword balance))
                    )
                  )
                )
                (assignmentOperator -=)
                (expression (number 24))
              )
            )
          )
        )
      )
      ;
    )
  )
)
/*======*/